On the GCC7 toolchain, we need to pass `-rpath-link` to the compiler rather
than just `-rpath`. This also works with the `toolchain_c` toolchain. The
needed feature can be tested by attempting to build the `pygobject` or
`poppler` feedstocks, which invoke this codepath in their builds.

diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index d10327c..0fc08f8 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -157,7 +157,7 @@ class CCompiler(object):
                         args.append('-rpath')
                         args.append(library_path)
                     else:
-                        args.append('-Wl,-rpath,' + library_path)
+                        args.append('-Wl,-rpath-link,' + library_path)
 
             runtime_paths.append(library_path)
 
